library/src/scripts/flyouts/NewPostMenu.tsx

```diff
@@ -59,47 +71,285 @@ function ActionItem({ item, style }: { item: IAddPost; style: ITransition }) {
+                break;
+            default:
+                // We don't want to lose focus in case we already have
+                if (state.buttonFocus) {
+                    dispatch({ type: "set_button_focus", focus: true });
+                }
+                break;
+        }
+    };
+
+    const onKeyDown = (event: React.KeyboardEvent<any>) => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        switch (event.key) {
+            case "Escape":
+                if (state.open) {
+                    dispatch({ type: "set_open", open: false });
+                    dispatch({ type: "set_button_focus", focus: true });
+                    dispatch({ type: "set_focused_item", item: undefined });
+                }
+                break;
+            case "Home":
+                if (state.open) {
+                    dispatch({ type: "set_focused_item", item: 0 });
+                }
+                break;
+            case "End":
+                if (state.open) {
+                    dispatch({ type: "set_focused_item", item: items.length - 1 });
+                }
+                break;
+            case "ArrowUp":
+                if (state.open && typeof state.focusedItem != "undefined") {
+                    dispatch({ type: "set_focused_item", item: (state.focusedItem + 1) % items.length });
+                }
+                break;
+            case "ArrowDown":
+                if (state.open && typeof state.focusedItem != "undefined") {
+                    dispatch({ type: "set_focused_item", item: (state.focusedItem - 1 + items.length) % items.length });
+                }
+                break;
+            case "Enter":
+                if (typeof state.focusedItem !== "undefined") {
+                    const item = items[state.focusedItem];
+                    switch (item.type) {
+                        case PostTypes.LINK:
+                            window.location.href = item.action as string;
+                            dispatch({ type: "set_open", open: false });
+                            dispatch({ type: "set_button_focus", focus: true });
+                            dispatch({ type: "set_focused_item", item: undefined });
+                            break;
+                        case PostTypes.BUTTON:
+                            (item.action as () => void)();
+                            dispatch({ type: "set_open", open: false });
+                            dispatch({ type: "set_button_focus", focus: true });
+                            dispatch({ type: "set_focused_item", item: undefined });
+                            break;
+                        default:
+                            break;
+                    }
+                }
+                break;
+            default:
+                const itemList = items
+                    .map((item, index) => ({ ...item, index }))
+                    .filter(item => item.label.startsWith(event.key));
```

Comment:
This may be part of your problem, you remember I told you to do a string to lower on your event key and on the label?

Reply:
This is related to the above comment.